src/ci/docker/host-x86_64/dist-various-2/build-fuchsia-toolchain.sh

```diff
@@ -3,52 +3,57 @@
 set -ex
 source shared.sh
 
-ZIRCON=e9a26dbc70d631029f8ee9763103910b7e3a2fe1
-
-mkdir -p zircon
-pushd zircon > /dev/null
-
-# Download sources
-git init
-git remote add origin https://github.com/rust-lang-nursery/mirror-google-fuchsia-zircon
-git fetch --depth=1 origin $ZIRCON
-git reset --hard FETCH_HEAD
-
-# Download toolchain
-./scripts/download-toolchain
-chmod -R a+rx prebuilt/downloads/clang+llvm-x86_64-linux
-cp -a prebuilt/downloads/clang+llvm-x86_64-linux/. /usr/local
-
-build() {
-  local arch="$1"
-
-  case "${arch}" in
-    x86_64) tgt="zircon-pc-x86-64" ;;
-    aarch64) tgt="zircon-qemu-arm64" ;;
-  esac
-
-  hide_output make -j$(getconf _NPROCESSORS_ONLN) $tgt
-  dst=/usr/local/${arch}-fuchsia
-  mkdir -p $dst
-  cp -a build-${tgt}/sysroot/include $dst/
-  cp -a build-${tgt}/sysroot/lib $dst/
+FUCHSIA_SDK=4xjxrGUrDbQ6_zJwj6cDN1IbWsWV5aCQXC_zO_Hu0XkC
+FUCHSIA_SDK_SHA256=e318f1ac652b0db43aff32708fa70337521b5ac595e5a0905c2ff33bf1eed179
+ZIRCON_SDK_USR_DIR=/usr/local/core-linux-amd64-fuchsia-sdk
+CLANG_ZIRCON=e9a26dbc70d631029f8ee9763103910b7e3a2fe1
+
+install_clang() {
+  # Other dists currently depend on this Clang... keeping only for creation and
+  #  symlinking of Clang
+  mkdir -p clang_build
+  pushd clang_build > /dev/null
+
+  # Download sources
+  git init
+  git remote add origin https://github.com/rust-lang-nursery/mirror-google-fuchsia-zircon
+  git fetch --depth=1 origin $CLANG_ZIRCON
+  git reset --hard FETCH_HEAD
+
+  # Download clang+llvm
+  ./scripts/download-toolchain
+  chmod -R a+rx prebuilt/downloads/clang+llvm-x86_64-linux
+  cp -a prebuilt/downloads/clang+llvm-x86_64-linux/. /usr/local
+
+  popd > /dev/null
+  rm -rf clang_build
 }
 
-# Build sysroot
-for arch in x86_64 aarch64; do
-  build ${arch}
-done
-
-popd > /dev/null
-rm -rf zircon
-
-for arch in x86_64 aarch64; do
-  for tool in clang clang++; do
-    cat >/usr/local/bin/${arch}-fuchsia-${tool} <<EOF
-#!/bin/sh
-${tool} --target=${arch}-fuchsia --sysroot=/usr/local/${arch}-fuchsia "\$@"
-EOF
-    chmod +x /usr/local/bin/${arch}-fuchsia-${tool}
+install_zircon_libs() {
+  mkdir -p zircon
+  pushd zircon > /dev/null
+
+  # Download Fuchsia SDK (with Zircon libs)
+  curl -LO \
+    "https://chrome-infra-packages.appspot.com/dl/fuchsia/sdk/core/linux-amd64/+/${FUCHSIA_SDK}"
+  echo "$(echo ${FUCHSIA_SDK_SHA256}) ${FUCHSIA_SDK}" | sha256sum --check --status
+  unzip -qq ${FUCHSIA_SDK} -d core-linux-amd64
+
+  # Moving SDK into Docker's user-space
+  mkdir -p ${ZIRCON_SDK_USR_DIR}
+  cp -r core-linux-amd64/* ${ZIRCON_SDK_USR_DIR}
+
+  for arch in x86_64 aarch64; do
+    # CFLAGS and CXXFLAGS env variables in main Dockerfile handle sysroot linking
+    for tool in clang clang++; do
+      ln -s /usr/local/bin/${tool} /usr/local/bin/${arch}-fuchsia-${tool}
+    done
+    ln -s /usr/local/bin/llvm-ar /usr/local/bin/${arch}-fuchsia-ar
```

Comment:
This is related to toolchain setup, not sysroot, so I'd expect this to be in `install_clang` and not `install_zircon_libs`.